File: app/Models/PurchaseReturnItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class PurchaseReturnItem extends Model
{
    use HasFactory;

    protected $fillable = [
        'purchase_return_id',
        'batch_id',
        'quantity',
        'purchase_price',
        'total',
    ];

    public function purchaseReturn()
    {
        return $this->belongsTo(PurchaseReturn::class);
    }

    public function batch()
    {
        return $this->belongsTo(Batch::class);
    }
}

File: desktop/lib/features/purchases/domain/entities/purchase_return_item.dart
class PurchaseReturnItem {
  final String? id;
  final String? purchaseReturnId;
  final String batchId;
  final String? batchNumber;
  final String? medicineName;
  final int quantity;
  final double purchasePrice;
  final double total;

  const PurchaseReturnItem({
    this.id,
    this.purchaseReturnId,
    required this.batchId,
    this.batchNumber,
    this.medicineName,
    required this.quantity,
    required this.purchasePrice,
    required this.total,
  });
}
